src/sbuf/MemBlob.cc

```diff
@@ -124,6 +124,17 @@ MemBlob::append(const char *source, const size_type n)
     ++Stats.append;
 }
 
+void
+MemBlob::moveToStartAndKeep(const size_type offset, const size_type n)
+{
+    if (n > 0) {
+        Must(offset);
```

Comment:
commit 486286c4fa

Reply:
After all the renaming, I was expecting something like this instead:

```C++
    if (n && size) {
        Must(LockCount() <= 1);
        Must(n <= size);
        size -= n;
        memmove(mem, mem + n, size);
    }
```

Reply:
I did this change (commit cfb9a669) however I think that the old scheme was more correct because we should not allow n higher than MemBlob::size. The old scheme azures that a zero MemBlob::size requires zero "n" parameter.

Not important however.

Reply:
Yes, it is not clear what "shift an empty container 100 bytes to the left" should mean. Some would say that it means "give me an empty container because shifting nothing changes nothing" and some would say "assert because you cannot shift nothing by anything except zero". Yet others might even insist that "you cannot shift nothing, not even by zero!".

For an integer shift operation, I think my interpretation is the best. For a _buffer_ shift operation, your interpretation is better because the caller should have no legitimate need to shift more data than there was in the buffer. Please move `Must(n <= size)` check outside the `if` statement.

Reply:
commit 351ed5c7